nyaa/forms.py

```diff
@@ -286,13 +286,120 @@ def validate_user_class(form, field):
             raise ValidationError('Please select a proper user class')
 
 
+class UserTorrentMassAction(FlaskForm):
+    action = SelectField('action', choices=[
+        ('hide', 'Hide'),
+        ('unhide', 'Unhide'),
+        ('remake', 'Remake'),
+        ('unremake', 'Unremake'),
+        ('move_category', 'Move To Category')
+    ])
+
+    category = StringField('category')
+
+    selected_category = {'main': None, 'sub': None}
+
+    admin_actions = [
+        ('anonymise', 'Make Anonymous'),
+        ('delete', 'Delete'),
+        ('unanonymise', 'Remove Anonymity')
+    ]
+
+    is_admin = False
+
+    def __init__(self, *args, **kwargs):
+        super(UserTorrentMassAction, self).__init__(*args, **kwargs)
+        self.selected_torrents = [x for x in kwargs.get('selected_torrents', []) if x.isdigit()]
+        user = kwargs.get('user', None)
+
+        if user is None:
+            raise TypeError('User needs to be passed as a keyword parameter.')
+
+        if user.is_moderator or user.is_superadmin:
+            self.is_admin = True
+            self.action.choices = self.action.choices + self.admin_actions
+
+    def validate(self, user=None):
+        super(UserTorrentMassAction, self).validate()
+
+        torrents = db.session.query(models.Torrent) \
+            .filter(models.Torrent.id.in_(self.selected_torrents)) \
+            .filter(models.Torrent.user == user).all()
+
+        if len(torrents) != len(self.selected_torrents):
+            return False
+
+        self.selected_torrents = torrents
+
+        if self.action.data == 'move_category':
+            [primary, secondary] = self.category.data.split('_')
```

Comment:
Oh, I think I know what happened. Did you try to move the torrent to a new category, but didn't change the default category option?